bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVaneerSettingsFactory.java

```diff
@@ -0,0 +1,312 @@
+  /**
+   * This method builds BigtableDataSettings#readModifyWriteSettngs when short timeout is other
+   * than 60_000 ms.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param rpcTimeoutMs a long value for RPC timeout.
+   */
+  private static void buildReadModifyWriteSettings(Builder builder, long rpcTimeoutMs) {
+    if(rpcTimeoutMs != SHORT_TIMEOUT_MS_DEFAULT) {
+      builder.readModifyWriteRowSettings()
+          .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+    }
+  }
+
+  /**
+   * This method builds BigtableDataSettings#checkAndMutateRowSettings when short timeout is other
+   * than 60_000 ms.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param rpcTimeoutMs a long value for RPC timeout.
+   */
+  private static void buildCheckAndMutateRowSettings(Builder builder, long rpcTimeoutMs) {
+    if(rpcTimeoutMs != SHORT_TIMEOUT_MS_DEFAULT) {
+      builder.checkAndMutateRowSettings()
+          .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+    }
+  }
+
+  /**
+   * To create default {@link RetrySettings} for all idempotent method.
+   *
+   * @param options a {@link BigtableOptions} object.
+   * @return an object of {@link RetrySettings} with idempotent method configuration.
+   */
+  private static RetrySettings buildIdempotentRetrySettings(BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+
+    RetrySettings.Builder retryBuilder = RetrySettings.newBuilder()
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+        .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries());
+
+    // configurations for RPC timeouts
+    Duration shortRpcTimeout = ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs());
+    retryBuilder
+        .setInitialRpcTimeout(shortRpcTimeout)
+        .setMaxRpcTimeout(shortRpcTimeout)
+        .setTotalTimeout(ofMillis(options.getCallOptionsConfig().getLongRpcTimeoutMs()));
+
+    if (retryOptions.allowRetriesWithoutTimestamp()) {
+      LOG.warn("Retries without Timestamp does not support yet.");
+    }
+
+    return retryBuilder.build();
+  }
+
+  /**
+   * To create {@link CredentialsProvider} based on {CredentialType of {@link BigtableOptions}.
+   *
+   * @param credentialOptions a {@link CredentialOptions} object.
+   * @throws IOException if any.
+   */
+  private static CredentialsProvider buildCredentialProvider(
+      CredentialOptions credentialOptions) throws IOException {
+    CredentialsProvider credentialsProvider = NoCredentialsProvider.create();
+
+    switch (credentialOptions.getCredentialType()) {
+      case DefaultCredentials:
+        credentialsProvider = BigtableStubSettings.defaultCredentialsProviderBuilder().build();
+        break;
+      case P12:
+      case SuppliedCredentials:
```

Comment:
should this be something along the lines of 
```
credential = ((UserSuppliedCredentialOptions)credentialOptions).getCredential();
credentialsProvider = FixedCredentialsProvider.create(credential);
```

Reply:
Updated [buildCredentialProvider()](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/1973/files#diff-3b1db200ad0c45bb25cfa85e47b21766R284) to use [CredentialFactory#getCredentials](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/CredentialFactory.java#L129-L156)